code/game/objects/structures/industrial_lift.dm

```diff
@@ -227,16 +230,36 @@ GLOBAL_LIST_EMPTY(lifts)
 		destination = get_step_multiz(src, going)
 	else
 		destination = going
+	if(istype(destination, /turf/closed/wall))
+		var/turf/closed/wall/C = destination
+		do_sparks(2, FALSE, C)
+		C.dismantle_wall(devastated = TRUE)
+		for(var/mob/M in urange(8, src))
+			shake_camera(M, 2, 3)
+		
+		playsound(C, 'sound/effects/meteorimpact.ogg', 100, TRUE)
 	if(going == DOWN)
 		for(var/mob/living/crushed in destination.contents)
 			to_chat(crushed, "<span class='userdanger'>You are crushed by [src]!</span>")
 			crushed.gib(FALSE,FALSE,FALSE)//the nicest kind of gibbing, keeping everything intact.
 	else if(going != UP) //can't really crush something upwards
-		for(var/obj/structure/anchortrouble in destination.contents)
-			if(!QDELETED(anchortrouble) && anchortrouble.anchored && (!istype(anchortrouble, /obj/structure/holosign)) && anchortrouble.layer >= GAS_PUMP_LAYER) //to avoid pipes, wires, etc
+		var/atom/throw_target = get_edge_target_turf(src, turn(going, pick(45, -45))) //finds a spot to throw the victim at for daring to be hit by a tram
+		for(var/obj/structure/victimstructure in destination.contents)
+			if(!QDELETED(victimstructure) && (!istype(victimstructure, /obj/structure/holosign)) && victimstructure.layer >= GAS_PUMP_LAYER) //to avoid wires
```

Comment:
Is there a check for incorporeal structures you can do that isn't an istype check?

Reply:
Seconding kyler's question, also you should continue if the object is QDELETED, rather then checking to see if it's not. See https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-early-return

It just makes things easier to read

Reply:
As far as I'm aware there isn't any real 'check' for incorporeal structures past something like making a new isincorporeal proc or something weird like checking to see if it's initially dense or w/e

Reply:
pain